project_task_materials/models/project.py

```diff
@@ -32,9 +18,19 @@ class Task(models.Model):
 class ProjectTaskMaterials(models.Model):
     _name = "project.task.materials"
     _description = "Task Materials Used"
+
     task_id = fields.Many2one(
         comodel_name='project.task', string='Task', ondelete='cascade',
         required=True)
     product_id = fields.Many2one(
         comodel_name='product.product', string='Product', required=True)
     quantity = fields.Float(string='Quantity')
+
+    @api.model
+    def create(self, values):
+        if 'quantity' in values:
+            if not values['quantity'] > 0.0:
+                raise exceptions.Warning(
+                    _('Quantity of materials consumed must be greater than 0.')
+                )
```

Comment:
Validations are is best done using `@api.constrains`.
`Warning` is [deprecated](https://github.com/odoo/odoo/blob/9.0/openerp/exceptions.py#L38).
Use `ValidationError`.